app/src/main/java/com/example/evently/ui/common/ConfirmFragmentNoInput.java

```diff
@@ -0,0 +1,112 @@
+import android.graphics.drawable.GradientDrawable;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.FrameLayout;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+
+import com.example.evently.R;
+
+public class ConfirmFragmentNoInput extends DialogFragment {
+
+    static final String headerKey = "header";
+    static final String messageKey = "message";
+    static final String cancelKey = "cancel";
+    static final String fragmentKey = "fragmentClass";
+
+    public ConfirmFragmentNoInput() {}
+
+    public static ConfirmFragmentNoInput newInstance(String headerText, String messageText, String cancelText, Class<? extends Fragment> fragmentClass) {
+        ConfirmFragmentNoInput confirmFragment = new ConfirmFragmentNoInput();
+        Bundle args = new Bundle();
+        args.putString(headerKey, headerText);
+        args.putString(messageKey, messageText);
+        args.putString(cancelKey, cancelText);
+        args.putString(fragmentKey, fragmentClass.getCanonicalName());
+        confirmFragment.setArguments(args);
+        return confirmFragment;
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_confirmation_no_user_input, container, false);
+        TextView header = view.findViewById(R.id.confirmation_no_user_input_header);
+        TextView message = view.findViewById(R.id.confirmation_no_user_input_message);
+        Button cancel = view.findViewById(R.id.confirmation_no_user_input_cancel_button);
+        Bundle args = getArguments();
+
+        if (args == null) throw new RuntimeException("Illegal createView Call made - ConfirmFragmentNoInput class");
+
+        header.setText(args.getString(headerKey));
+        message.setText(args.getString(messageKey));
+        cancel.setText(args.getString(cancelKey, "Cancel"));
+
+        cancel.setOnClickListener(v -> dismiss());
+
+        return view;
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        Bundle args = getArguments();
+
+        if (args == null) throw new RuntimeException("Illegal createView Call made - ConfirmFragmentNoInput class");
+        Fragment actionFragment;
+        try {
+            actionFragment = (Fragment) Class.forName(args.getString(fragmentKey)).newInstance();
+        } catch (ClassNotFoundException | IllegalAccessException | java.lang.InstantiationException e) {
+            throw new RuntimeException(e);
+        }
+
+        getChildFragmentManager().registerFragmentLifecycleCallbacks(
```

Comment:
This might be the old way to do these things. Thew modern way is to just use`onViewCreated` in the fragment you're trying to implement the logic for.